### ISSUE TITLE
Stream::getMetadata()

### ISSUE THREAD

**Issue by weierophinney** [laminas/laminas-diactoros#22]
What must return Stream::getMetadata() if resource is detached?

P.S.: 
What should be the level of expectation of the error?
if i use:

```
$stream->write(function () {});
```

Do I have to expects an RutimeException?

---

Originally posted by @easy-system at https://github.com/zendframework/zend-diactoros/issues/168

**Comment by Xerkus** [comment]
On detached resource `getMetadata()` should return null as there is no metadata to be had.

**Comment by boesing** [comment]
@Xerkus as per PSR-7, `getMetadata` without a specific argument has to return an associative array. I'd say, in case of detached resources, that should be just an empty array, no?

**Comment by Xerkus** [comment]
@boesing you are right. It should be empty array. I misread the docblock. Null is only expected when specific key is not found.
